lib/api/core/entrypoints/embedded/index.js

```diff
@@ -0,0 +1,464 @@
+      url = `/${request.input.controller}/${request.input.action}`;
+      if (request.input.resource.index) {
+        url += '/' + request.input.resource.index;
+      }
+      if (request.input.resource.collection) {
+        url += '/' + request.input.resource.collection;
+      }
+      if (request.input.resource._id) {
+        url += '/' + request.input.resource._id;
+      }
+
+      const queryString = Object.keys(request.input.args)
+        .map(k => {
+          let arg = k + '=';
+          const val = request.input.args[k];
+
+          arg += typeof val === 'object' ? JSON.stringify(val) : val;
+
+          return arg;
+        })
+        .join('&');
+
+      if (queryString !== '') {
+        url += '?' + queryString;
+      }
+
+      if (request.input.jwt) {
+        try {
+          const
+            b64Paylod = request.input.jwt.split('.')[1],
+            payload = new Buffer(b64Paylod, 'base64').toString('utf8');
+          user = JSON.parse(payload)._id;
+        }
+        catch (err) {
+          this.kuzzle.pluginsManager.trigger('log:warn', 'Unable to extract user from jwt token: ' + request.input.jwt);
+        }
+      }
+    }
+
+    this.logger.info([
+      connection.ips[connection.ips.length - 1 - this.config.logs.accessLogIpOffset],
+      '-',
+      user || '-',
+      '[' + moment().format('DD/MMM/YYYY:HH:mm:ss ZZ') + ']',
+      `"${verb} ${url} ${protocol}"`,
+      request.status || '-',
+      request.response ? Buffer.byteLength(JSON.stringify(request.response)) : '-',
+      connection.headers.referer ? `"${connection.headers.referer}"` : '-',
+      connection.headers['user-agent'] ? `"${connection.headers['user-agent']}"` : '-'
+    ].join(' '));
+  }
+
+  /*
+  -----------------------------------------------------------------------
+  methods exposed to protocols
+  -----------------------------------------------------------------------
+  */
+
+  /**
+   *
+   * @param {Request} request
+   * @param cb
+   */
+  execute (request, cb) {
+    this.kuzzle.funnel.execute(request, (error, result) => {
+      if (error && !result.error) {
+        result.setError(error);
+      }
+
+      this.logAccess(result);
+
+      const response = result.response.toJSON();
+
+      cb(this.constructor._removeErrorStack(response));
+    });
+  }
+
+  /**
+   *
+   * @param {ClientConnection} connection
+   */
+  newConnection (connection) {
+    this.clients[connection.id] = connection;
+    this.kuzzle.router.newConnection(new RequestContext({
+      connectionId: connection.id,
+      protocol: connection.protocol,
+      headers: connection.headers
+    }));
+  }
+
+  /**
+   * @param {string} connectionId
+   */
+  removeConnection (connectionId) {
+    if (!this.clients[connectionId]) {
+      return;
+    }
+
+    const connection = this.clients[connectionId];
+    this.kuzzle.router.removeConnection(new RequestContext({
+      connectionId: connectionId,
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use shorthand for property "connectionId". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS3498)